Tools/WinMLRunner/BindingUtilities.h

```diff
@@ -144,6 +144,35 @@ namespace BindingUtilities
         return elementStrings;
     }
 
+    template <typename T>
+    void PreProcessImageToBinding(const SoftwareBitmap& softwareBitmap, ModelBinding<T>& binding, float scale, const std::array<float, 3>& meanStdDev)
+    {
+        const auto imgHeight = softwareBitmap.PixelHeight();
+        const auto imgWidth = softwareBitmap.PixelWidth();
+        const auto resultSize = imgHeight * imgWidth * 3;
+        if (binding.GetDataBufferSize() != resultSize)
+        {
+            throw hresult_invalid_argument(L"Image Input size/shape is different from what model expects");
+        }
+
+        T* bindingData = binding.GetData();
+        if (scale != 1.0f &&
+            (meanStdDev[0] != 0 || meanStdDev[1] != 0 || meanStdDev[2] != 0))
+        {
+            Buffer sbBuffer(imgHeight * imgWidth * 4);
+            softwareBitmap.CopyToBuffer(sbBuffer);
+            uint8_t *sbBufferData = sbBuffer.data();
+
+            //Roll the array correctly for the tensor
+            for (int i = 0, count = 0; i < imgHeight * imgWidth; ++i, count += 4)
```

Comment:
For inputDataType imageBGR, the order of meanStdDev doesn't match.

Reply:
Yes, you can reorder them to RGB. Just make sure they are consistent. And you don't need to worry about the pixelFormat the model wants because Winml will handle that.

Reply:
My understanding is that WinML would only reorder the inputs if the input data was the VideoFrame datatype.  These are being passed to WinML as a tensor, so I'm assuming that I will need to reorder the channels if it's required by the model in this code.